perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_virt_adGroupNameTest.java

```diff
@@ -0,0 +1,92 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.Group;
+import cz.metacentrum.perun.core.bl.AttributesManagerBl;
+import cz.metacentrum.perun.core.bl.GroupsManagerBl;
+import cz.metacentrum.perun.core.bl.PerunBl;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class urn_perun_group_attribute_def_virt_adGroupNameTest {
+
+	private static final String defAdGroupNameAttributeName = AttributesManager.NS_GROUP_ATTR_DEF + ":adGroupName";
+
+	private final urn_perun_group_attribute_def_virt_adGroupName classInstance = new urn_perun_group_attribute_def_virt_adGroupName();
+	private final AttributeDefinition adDGroupNameDefinition = classInstance.getAttributeDefinition();
+
+	private final Group groupA = setUpGroup(1, null, "groupA", "A");
+	private final Group groupB = setUpGroup(2, 1, "groupB", "B");
+	private final Group groupC = setUpGroup(3, 2, "groupC", "C");
+
+	private final Attribute attributeA = new Attribute();
+	private final Attribute attributeB = new Attribute();
+	private final Attribute attributeC = new Attribute();
+
+	private PerunSessionImpl sess;
+
+	@Before
+	public void setUp() throws Exception {
+		attributeA.setValue("A");
+		attributeB.setValue("B");
+		attributeC.setValue("C");
+
+		//prepare mocks
+		sess = mock(PerunSessionImpl.class);
+		PerunBl perunBl = mock(PerunBl.class);
+		AttributesManagerBl am = mock(AttributesManagerBl.class);
+		GroupsManagerBl gm = mock(GroupsManagerBl.class);
+
+		when(sess.getPerunBl())
+			.thenReturn(perunBl);
+		when(perunBl.getAttributesManagerBl())
+			.thenReturn(am);
+		when(perunBl.getGroupsManagerBl())
+			.thenReturn(gm);
+
+		when(gm.getParentGroup(sess, groupB))
+			.thenReturn(groupA);
+		when(gm.getParentGroup(sess, groupC))
+			.thenReturn(groupB);
+
+		when(am.getAttribute(eq(sess), eq(groupA), any(String.class)))
```

Comment:
You should probably test here that the given attribute name is correct.

Reply:
Fixed